app/channel_share.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/store"
+)
+
+// SharedChannels
+
+func (a *App) SaveSharedChannel(sc *model.SharedChannel) (*model.SharedChannel, error) {
+	// Check channel is shared.
+	_, err := a.GetSharedChannel(sc.ChannelId)
+	if err != nil {
+		return nil, errors.New("channel is already shared.")
+	}
```

Comment:
Thanks @streamer45 for another great review.  

The logic here was actually wrong.  If `err == nil` then the channel is already shared.  This was masked by the fact the correct logic was also in the slash command and didn't get this far if the channel is already shared.  

It's possible that an already shared channel will return an err from `GetSharedChannel`, such as DB connection problem, but it will still fail further along with a more cryptic `save_shared_channel: ChannelId=eanhnuapeb84inkq34qxbactio: pq: duplicate key value violates unique constraint "sharedchannels_pkey"`.

As for translations, I'll reach out to our PM and see what the expectation is.

Reply:
Our PM indicated we won't do translations for the MVP.